### PR TITLE
Add an isButton flag to the Semantic widget,use it for MaterialButton

### PR DESCRIPTION
**Pull request by amirh**
Will follow-up with another CL for Cupertino buttons.

This can only land after we roll the engine with https://github.com/flutter/engine/pull/4254
(Expect tests to fail until the roll)

### CODE REVIEW

File: packages/flutter/lib/src/rendering/proxy_box.dart
Comment (Hixie) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: I think this documentation may have a grammar error in it.

File: packages/flutter/lib/src/widgets/basic.dart
Comment (Hixie) [inline packages/flutter/lib/src/widgets/basic.dart]: this should probably elaborate on what "a button" means.

**Comment by Hixie** [comment]
LGTM modulo some minor documentation issues

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/rendering/proxy_box.dart
Comment (amirh) [reply packages/flutter/lib/src/rendering/proxy_box.dart]: done

File: packages/flutter/lib/src/widgets/basic.dart
Comment (amirh) [reply packages/flutter/lib/src/widgets/basic.dart]: I'm not sure how to elaborate what a button means, I could try to describe it as something tap-able,  but not everything that's tap-able should have this flag set to true.

Instead I added some documentation of how this affects TalkBack/VoiceOver (provides a "button" hint)

**Comment by amirh** [comment]
Will merge once an engine roll lands

**Comment by Hixie** [comment]
We should also add these semantics to the cupertino buttons.

**Pull request merged**
